features/dd-sdk-android-flags/src/test/kotlin/com/datadog/android/flags/internal/net/PrecomputedAssignmentsDownloaderTest.kt

```diff
@@ -0,0 +1,383 @@
+
+    @Mock
+    lateinit var mockCall: Call
+
+    private lateinit var testedDownloader: PrecomputedAssignmentsDownloader
+    private lateinit var fakeFlagsContext: FlagsContext
+    private lateinit var fakeEvaluationContext: EvaluationContext
+
+    @BeforeEach
+    fun `set up`(forge: Forge) {
+        fakeFlagsContext = FlagsContext(
+            clientToken = forge.anAlphabeticalString(),
+            applicationId = forge.anAlphabeticalString(),
+            site = DatadogSite.US1,
+            env = "test"
+        )
+
+        fakeEvaluationContext = EvaluationContext(
+            targetingKey = forge.anAlphabeticalString(),
+            attributes = mapOf("plan" to "premium")
+        )
+
+        testedDownloader = PrecomputedAssignmentsDownloader(
+            callFactory = mockCallFactory,
+            internalLogger = mockInternalLogger,
+            flagsContext = fakeFlagsContext,
+            requestFactory = mockRequestFactory
+        )
+    }
+
+    // region readPrecomputedFlags() - Success cases
+
+    @Test
+    fun `M return response body W readPrecomputedFlags() { successful request }`(
+        @StringForgery fakeResponseBody: String,
+        @StringForgery(regex = "https://[a-z]+\\.(com|net)/[a-z]+") fakeUrl: String
+    ) {
+        // Given
+        val fakeRequest = Request.Builder()
+            .url(fakeUrl)
+            .build()
+        val fakeResponse = createSuccessfulResponse(fakeResponseBody, fakeUrl)
+
+        whenever(mockRequestFactory.create(fakeEvaluationContext, fakeFlagsContext))
+            .doReturn(fakeRequest)
+        whenever(mockCallFactory.newCall(fakeRequest)).doReturn(mockCall)
+        whenever(mockCall.execute()).doReturn(fakeResponse)
+
+        // When
+        val result = testedDownloader.readPrecomputedFlags(fakeEvaluationContext)
+
+        // Then
+        assertThat(result).isEqualTo(fakeResponseBody)
+        verify(mockRequestFactory).create(fakeEvaluationContext, fakeFlagsContext)
+    }
+
+    @Test
+    fun `M call factory with correct parameters W readPrecomputedFlags()`(
+        @StringForgery fakeResponseBody: String,
+        @StringForgery(regex = "https://[a-z]+\\.(com|net)/[a-z]+") fakeUrl: String
+    ) {
+        // Given
+        val fakeRequest = Request.Builder()
+            .url(fakeUrl)
+            .build()
+        val fakeResponse = createSuccessfulResponse(fakeResponseBody, fakeUrl)
+
+        whenever(mockRequestFactory.create(fakeEvaluationContext, fakeFlagsContext))
+            .doReturn(fakeRequest)
+        whenever(mockCallFactory.newCall(fakeRequest)).doReturn(mockCall)
+        whenever(mockCall.execute()).doReturn(fakeResponse)
+
+        // When
+        testedDownloader.readPrecomputedFlags(fakeEvaluationContext)
+
+        // Then
+        argumentCaptor<EvaluationContext> {
+            verify(mockRequestFactory).create(capture(), eq(fakeFlagsContext))
+            assertThat(lastValue).isEqualTo(fakeEvaluationContext)
+        }
```

Comment:
seems like argument captor is not really needed here and everything can be written simply as:

```suggestion
        verify(mockRequestFactory).create(fakeEvaluationContext, fakeFlagsContext)
```